designer/restmodel/src/main/scala/pl/touk/nussknacker/restmodel/definition/package.scala

```diff
@@ -73,56 +75,49 @@ package object definition {
 
   @JsonCodec(encodeOnly = true) final case class UISourceParameters(sourceId: String, parameters: List[UIParameter])
 
-  final case class NodeEdges(
+  final case class UINodeEdges(
       componentId: ComponentInfo,
       edges: List[EdgeType],
       canChooseNodes: Boolean,
       isForInputDefinition: Boolean
   )
 
-  object NodeEdges {
+  object UINodeEdges {
     implicit val componentIdEncoder: Encoder[ComponentInfo] = Encoder.encodeString.contramap(_.toString)
 
-    implicit val encoder: Encoder[NodeEdges] = deriveConfiguredEncoder
+    implicit val encoder: Encoder[UINodeEdges] = deriveConfiguredEncoder
   }
 
-  object ComponentNodeTemplate {
+  object UIComponentNodeTemplate {
 
     def create(
         componentInfo: ComponentInfo,
         nodeTemplate: NodeData,
-        categories: List[String],
         branchParametersTemplate: List[NodeParameter]
-    ): ComponentNodeTemplate =
-      ComponentNodeTemplate(
+    ): UIComponentNodeTemplate =
+      UIComponentNodeTemplate(
         componentInfo.`type`,
         componentInfo.name,
         nodeTemplate,
-        categories,
         branchParametersTemplate
       )
 
   }
 
-  @JsonCodec(encodeOnly = true) final case class ComponentNodeTemplate(
+  @JsonCodec(encodeOnly = true) final case class UIComponentNodeTemplate(
       // This field is used to generate unique key in DOM model on FE side (the label isn't unique)
```

Comment:
`NodeData.id` is a template (initial) `nodeId`. I have a plan to replace the `type` with `ComponentInfo`. But before I do it, it will be renamed to `ComponentId`